neqo-qpack/src/encoder.rs

```diff
@@ -257,28 +271,44 @@ impl QPackEncoder {
         }
     }
 
-    fn change_capacity(&mut self, value: u64) -> Res<()> {
+    fn need_change_capacity(&mut self, value: u64) {
         qdebug!([self], "change capacity: {}", value);
-        self.table.set_capacity(value)?;
-        EncoderInstruction::Capacity { value }.marshal(&mut self.send_buf, self.use_huffman);
+        self.change_capacity = Some(value);
+    }
+
+    fn maybe_send_change_capacity(&mut self, conn: &mut Connection, stream_id: u64) -> Res<()> {
+        if let Some(cap) = self.change_capacity.take() {
```

Comment:
If this write fails, then you won't send the capacity change.  So you should defer clearing the value until the write succeeds.